components/alert/alert.js

```diff
@@ -0,0 +1,153 @@
+		return [bodyCompactStyles, css`
+
+			:host {
+				animation: 600ms ease drop-in;
+				background: white;
+				border-radius: 0.3rem;
+				border: 1px solid var(--d2l-color-mica);
+				box-sizing: border-box;
+				display: flex;
+				flex: 1;
+				max-width: 710px;
+				width: 100%;
+			}
+
+			:host([hidden]) {
+				display: none;
+			}
+
+			.d2l-alert-highlight {
+				border-radius: 0.3rem 0 0 0.3rem;
+				bottom: 0;
+				left: 0;
+				margin: -1px;
+				min-width: 0.3rem;
+				top: 0;
+				width: 0.3rem;
+			}
+			:host([dir="rtl"]) .d2l-alert-highlight {
+				border-radius: 0 0.3rem 0.3rem 0;
+				left: auto;
+				right: 0;
+			}
+			:host([type="critical"]) .d2l-alert-highlight,
+			:host([type="error"]) .d2l-alert-highlight {
+				background-color: var(--d2l-color-feedback-error);
+			}
+			:host([type="warning"]) .d2l-alert-highlight {
+				background-color: var(--d2l-color-feedback-warning);
+			}
+			:host([type="default"]) .d2l-alert-highlight,
+			:host([type="call-to-action"]) .d2l-alert-highlight {
+				background-color: var(--d2l-color-feedback-action);
+			}
+			:host([type="success"]) .d2l-alert-highlight {
+				background-color: var(--d2l-color-feedback-success);
+			}
+
+			.d2l-alert-text {
+				flex: 1;
+				padding: 1rem 1.2rem 1rem 1.5rem;
+				position: relative;
+			}
+			:host([dir="rtl"]) .d2l-alert-text {
+				padding-left: 1.2rem;
+				padding-right: 1.5rem;
+			}
+			.d2l-alert-subtext {
+				margin: 0.5rem 0 0;
+			}
+
+			.d2l-alert-action {
+				margin: 0.3rem;
+			}
+
+			@keyframes drop-in {
+				from {
+					opacity: 0.0;
+					transform: translate(0,-10px);
+				}
+				to {
+					opacity: 1;
+					transform: translate(0,0);
+				}
+			}
+
+			@media (prefers-reduced-motion: reduce) {
+				:host {
+					animation: none;
+				}
+			}
+		`];
+	}
+
+	static get resources() {
+		return {
+			'ar': { 'close': 'إغلاق التنبيه'},
+			'de': { 'close': 'Benachrichtigung schließen' },
+			'en': { 'close': 'Close Alert'},
+			'es': { 'close': 'Alerta de cierre' },
+			'fr': { 'close': 'Fermer l\'alerte' },
+			'ja': { 'close': 'アラートを閉じる' },
+			'ko': { 'close': '경보 닫기' },
+			'nl': { 'close': 'Waarschuwing sluiten' },
+			'pt': { 'close': 'Fechar Alerta' },
+			'sv': { 'close': 'Stängningsvarning' },
+			'tr': { 'close': 'Kapatma Uyarısı' },
+			'zh-tw': { 'close': '關閉警示' },
+			'zh': { 'close': '关闭警报' }
+		};
+	}
+
+	constructor() {
+		super();
+		this.hasCloseButton = false;
+		this.type = 'default';
+	}
+
+	close() {
+		this.hidden = true;
+		this.dispatchEvent(new CustomEvent(
+			'd2l-alert-closed', { bubbles: true, composed: true }
+		));
+	}
+
+	_onButtonClick() {
+		this.dispatchEvent(new CustomEvent(
+			'd2l-alert-button-pressed', { bubbles: true, composed: true }
+		));
+	}
+
+	render() {
+		return html`
+			<div class="d2l-alert-highlight"></div>
+			<div class="d2l-alert-text">
+				<slot></slot>
+				${this.subtext ? html`<p class="d2l-body-compact d2l-alert-subtext">${this.subtext}</p>` : null}
```

Comment:
While `null` and `undefined` generally work, [the docs](https://lit-html.polymer-project.org/guide/writing-templates#rendering-nothing) suggest `''` is more consistent across browsers.  I haven't noticed anything specific, just going by what the docs say.  

The `nothing` sentinel is even better since really does render nothing at all, but requires importing it, and is only really necessary when rendering slot content where we want fallback slot content to be rendered when there is "nothing" in the slot.